src/interfaces/IXSLTProcessor.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Interface for SaxonJS functionality to enable dependency injection
+ * and future replacement of the SaxonJS library.
+ */
+export interface IXSLTProcessor {
+  /**
+   * Parse XML text into a document object
+   */
+  parse(options: {
+    text: string;
+  }): Promise<any>;
+
+  /**
+   * Evaluate XPath expression on a document
+   */
+  query(xpath: string, document: any): any;
+
+  /**
+   * Transform XML using XSLT
+   */
+  transform(
+    stylesheet: string, // is this the same as stylesheetLocation ?
```

Comment:
Yes, it is the same as SaxonJS's `stylesheetLocation`. If you think the name is unclear, let's rename it to `stylesheetUri` to remain consistent with `PlayerOptions`.

Reply:
Sorry I didn't clean up after myself. I have a tendency to take notes directly into the code.